core/src/main/scala/org/apache/spark/sql/delta/stats/DataSkippingReader.scala

```diff
@@ -426,9 +427,12 @@ trait DataSkippingReaderBase
       }
     case EqualTo(v: Literal, a) =>
       constructDataFilters(EqualTo(a, v))
+    case EqualNullSafe(v: Literal, a) =>
+      constructDataFilters(EqualNullSafe(a, v))
 
     // Match any file whose min/max range contains anything other than the rejected point.
-    case Not(EqualTo(SkippingEligibleColumn(a), SkippingEligibleLiteral(v))) =>
+    case Not(Equality(SkippingEligibleColumn(a), lit @ SkippingEligibleLiteral(v)))
```

Comment:
This change is not correct. See my comment below in the unit test.

Reply:
Good catch!